internal/cmd/controller/gitops/reconciler/gitjob_controller.go

```diff
@@ -104,6 +104,17 @@ func (r *GitJobReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	}
 
 	if errors.IsNotFound(err) && gitRepo.Status.Commit != "" {
+		if err := r.validateExternalSecretExist(ctx, &gitRepo); err != nil {
+			nsname := types.NamespacedName{Namespace: gitRepo.Namespace, Name: gitRepo.Name}
+			return ctrl.Result{}, grutil.UpdateErrorStatus(ctx, r.Client, nsname, gitRepo.Status, err)
+		}
+		logger.V(1).Info("Creating Git job resources")
+		if err := r.createJobRBAC(ctx, &gitRepo); err != nil {
+			return ctrl.Result{}, err
```

Comment:
nit: in the same way that we format errors when failing to create a git job (or even better: using the `%w` [placeholder](https://pkg.go.dev/fmt#Errorf) instead of `%v`), we could format errors happening when failing to create RBAC resources or the configmap, to make them easier to understand for users, eg:
```suggestion
			return ctrl.Result{}, fmt.Errorf("failed to create RBAC resources for git job: %w", err)
```

Reply:
yup, I've changed as suggested, and also... as I didn't want to have `%v`s and `%w`s in the same file... I ended up changing all `%v`s.